packages/odata-common/src/request-builder/batch/batch-request-serializer.ts

```diff
@@ -65,10 +65,19 @@ export function serializeRequest(
   return [
     'Content-Type: application/http',
     'Content-Transfer-Encoding: binary',
-    ...(method !== 'GET' ? [`Content-Id: ${uuid()}`] : []),
+    ...(method !== 'GET'
+      ? [
+          `Content-Id: ${
+            request.requestConfig.contentIdBatch?.header || uuid()
+          }`
+        ]
+      : []), // make function with logging nicer - is this needed here?.
     '',
     `${method} ${getUrl(odataRequest, options.subRequestPathType)} HTTP/1.1`,
     ...(requestHeaders.length ? requestHeaders : ['']),
+    ...(request.requestConfig.contentIdBatch?.etag
+      ? [`If-Match: ${request.requestConfig.contentIdBatch?.etag}`]
+      : []),
```

Comment:
[pp] This function needs to be polished in the real implementation for better readability.

Reply:
yes, I leave a comment. Also there should be a log statement if the `eTag` was also given in the header already.